host.go

```diff
@@ -695,4 +698,202 @@ func (h *Host) InitCommands(c *chat.Commands) {
+		"reload: re-read the allowlist file and keep or discard entries in the current allowlist but not in the file",
+		"reverify: kick all users not in the allowlist if allowlisting is enabled",
+		"status: show status information",
+	}
+
+	allowlistImport := func(args []string) (msgs []string, err error) {
+		var since time.Duration
+		if len(args) > 0 {
+			since, err = time.ParseDuration(args[0])
+			if err != nil {
+				return
+			}
+		}
+		cutoff := time.Now().Add(-since)
+		noKeyUsers := []string{}
+		forConnectedUsers(func(user *chat.Member, pk ssh.PublicKey) error {
+			if user.Joined().Before(cutoff) {
+				if pk == nil {
+					noKeyUsers = append(noKeyUsers, user.Identifier.Name())
+				} else {
+					h.auth.Whitelist(pk, 0)
+				}
+			}
+			return nil
+		})
+		if len(noKeyUsers) != 0 {
+			msgs = []string{fmt.Sprintf("users without a public key: %v", noKeyUsers)}
+		}
+		return
+	}
+
+	allowlistReload := func(args []string) error {
+		if !(len(args) > 0 && (args[0] == "keep" || args[0] == "flush")) {
+			return errors.New("must specify whether to keep or flush current entries")
+		}
+		if args[0] == "flush" {
+			h.auth.whitelist.Clear()
+		}
+		return h.auth.LoadWhitelistFromFile(h.auth.whitelistFile)
+	}
+
+	allowlistReverify := func(room *chat.Room) []string {
+		if !h.auth.WhitelistMode() {
+			return []string{"allowlist is disabled, so nobody will be kicked"}
+		}
+		var kicked []string
+		forConnectedUsers(func(user *chat.Member, pk ssh.PublicKey) error {
+			if h.auth.CheckPublicKey(pk) != nil && !user.IsOp { // we do this check here as well for ops without keys
+				kicked = append(kicked, user.Name())
+				user.Close()
+			}
+			return nil
+		})
+		if kicked != nil {
+			room.Send(message.NewAnnounceMsg(fmt.Sprintf("%v were kicked during pubkey reverification.", kicked)))
+		}
+		return nil
+	}
+
+	allowlistStatus := func() (msgs []string) {
+		if h.auth.WhitelistMode() {
+			msgs = []string{"The allowlist is currently enabled."}
+		} else {
+			msgs = []string{"The allowlist is currently disabled."}
+		}
+		whitelistedUsers := []string{}
+		whitelistedKeys := []string{}
+		h.auth.whitelist.Each(func(key string, item set.Item) error {
+			keyFP := item.Key()
+			if forConnectedUsers(func(user *chat.Member, pk ssh.PublicKey) error {
+				if pk != nil && sshd.Fingerprint(pk) == keyFP {
+					whitelistedUsers = append(whitelistedUsers, user.Name())
+					return errors.New("not an actual error, but exit early because we found the key")
+				}
+				return nil
+			}) == nil {
+				// if we land here, the key matches no users
+				whitelistedKeys = append(whitelistedKeys, keyFP)
+			}
+			return nil
+		})
+		if len(whitelistedUsers) != 0 {
+			msgs = append(msgs, fmt.Sprintf("The following connected users are on the allowlist: %v", whitelistedUsers))
+		}
+		if len(whitelistedKeys) != 0 {
+			msgs = append(msgs, fmt.Sprintf("The following keys of not connected users are on the allowlist: %v", whitelistedKeys))
```

Comment:
```suggestion
			msgs = append(msgs, fmt.Sprintf("Connected user keys not included on the allowlist: %v", whitelistedKeys))
```

Reply:
This is actually different: These are keys with ``inAllowlist && !userConnected``. (I read your suggestion as ``!inAllowlist && userConncted``)

Reply:
If we switch the loops, we can get connected unlisted users instead (we can also get both, but that means a bit more bookkeeping). What do you prefer?